molt/src/commands.rs

```diff
@@ -1150,6 +1150,39 @@ pub fn cmd_string_first(_interp: &mut Interp, _: ContextID, argv: &[Value]) -> M
     molt_ok!(pos)
 }
 
+/// string last *needleString* *haystackString* ?*lastIndex*?
+pub fn cmd_string_last(_interp: &mut Interp, _: ContextID, argv: &[Value]) -> MoltResult {
+    check_args(2, argv, 4, 5, "needleString haystackString ?lastIndex?")?;
+
+    let needle = argv[2].as_str();
+    let haystack = argv[3].as_str();
+
+    let last: Option<usize> = if argv.len() == 5 {
+        let arg = argv[4].as_int()?;
+
+        if arg < 0 {
+            None
+        } else if arg as usize >= haystack.len() {
+            Some(haystack.len() - 1)
+        } else {
+            Some(arg as usize)
+        }
+    } else {
+        Some(haystack.len() - 1)
+    };
+
+    let pos = match last {
+        Some(offset) =>
+            haystack[..=offset]
```

Comment:
Same issue here as with `string first`; `start` is indexing into the bytes of the string, not the (possibly multi-byte) characters.